### PR TITLE
gh-86179: Implement realpath() on Windows during getpath.py calculations

### PR DESCRIPTION
**Pull request by zooba**
* Issue: gh-86179

### CODE REVIEW

File: Modules/getpath.c [hunk 1]
```diff
@@ -502,6 +502,45 @@ getpath_realpath(PyObject *Py_UNUSED(self) , PyObject *args)
...
+        len = GetFinalPathNameByHandleW(hFile, resolved, MAXPATHLEN, VOLUME_NAME_DOS);
+        err = len ? 0 : GetLastError();
+        CloseHandle(hFile);
+    } else {
+        err = GetLastError();
+    }
+    Py_END_ALLOW_THREADS
+
+    if (err) {
+        return PyErr_SetFromWindowsErr(GetLastError());
```
Comment (pfmoore) [inline Modules/getpath.c]: This return leaks memory for `path`.

Comment (zooba) [reply Modules/getpath.c]: And it doesn't use the saved error number... thanks!

**Comment by zooba** [comment]
The failure is because the `realpath` implementation will convert drive names to uppercase, which causes case-sensitive path comparisons in the tests to fail. I'm pretty sure this was reported somewhere else for `os.path.realpath` as well, so I guess we need to do one more comparison before returning in both cases and not return case-only changes.

**Comment by Yhg1s** [comment]
This seems to have introduced a new test failure, but only when running tests in an installed Python _and_ using the unversioned binary (bin/python3 rather than bin/python3.13). Rolling back the changes in this PR makes both tests pass.

```
% /tmp/testinstall/bin/python3.13 -m test test_venv
Using random seed: 3548105807
0:00:00 load avg: 0.25 Run 1 test sequentially
0:00:00 load avg: 0.25 [1/1] test_venv

== Tests result: SUCCESS ==

1 test OK.

Total duration: 3.6 sec
Total tests: run=29 skipped=3
Total test files: run=1/1
Result: SUCCESS
```

```
% /tmp/testinstall/bin/python3 -m test test_venv
Using random seed: 3358697499
0:00:00 load avg: 0.06 Run 1 test sequentially
0:00:00 load avg: 0.06 [1/1] test_venv
test test_venv failed -- Traceback (most recent call last):
  File "/tmp/testinstall/lib/python3.13/test/test_venv.py", line 332, in test_sysconfig_symlinks
    self.assertEqual(out.strip(), expected, err)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: '/tmp/testinstall/bin/python3.13' != '/tmp/testinstall/bin/python3'
- /tmp/testinstall/bin/python3.13
?                             ---
+ /tmp/testinstall/bin/python3
 :

test_venv failed (1 failure)

== Tests result: FAILURE ==

1 test failed:
    test_venv

Total duration: 3.7 sec
Total tests: run=29 failures=1 skipped=3
Total test files: run=1/1 failed=1
Result: FAILURE
```

**Comment by zooba** [comment]
Rather than rolling back the (primarily Windows-only) change, better to skip the test.

I'll need to take some time to learn how venv/symlinks work on POSIX in order to diagnose and fix. As it's just a test, it may be faster if we can pull in someone who already knows that side of things.

**Comment by FFY00** [comment]
You can ping me on Discord, or an issue to discuss the POSIX behaviors.

**Comment by zooba** [comment]
I'll push a PR to skip that part of the test so it's not blocking the release. If we want to test venv later, we can figure it out then.

**Pull request merged**